.github/workflows/linter.yml

```diff
@@ -0,0 +1,53 @@
+on:
+  # For all pushes run the tests
+  push:
+  # Also trigger on pull_request
+  pull_request:
+  # Run nightly tests at 04:00 UTC
+  schedule:
+    - cron: "0 4 * * *"
+  # Manually triggered
+  workflow_dispatch:
+
+name: Lint with Flake8
```

Comment:
I would simply name the workflow 'Lint' (short and succinct).
This is the display name that appears in the GitHub worflows GUI.